lxd/storage_volumes.go

```diff
@@ -0,0 +1,466 @@
+	}
+
+	// Validate the requested storage volume configuration.
+	err = storageVolumeValidateConfig(poolName, req.VolumeConfig, poolStruct)
+	if err != nil {
+		return BadRequest(err)
+	}
+
+	// Create the database entry for the storage volume.
+	_, err = dbStoragePoolVolumeCreate(d.db, req.VolumeName, volumeType, poolID, req.VolumeConfig)
+	if err != nil {
+		return InternalError(fmt.Errorf("Error inserting %s of type %s into database: %s", poolName, req.VolumeType, err))
+	}
+
+	s, err := storagePoolVolumeInit(d, poolName, req.VolumeName, volumeType)
+	if err != nil {
+		return InternalError(err)
+	}
+
+	// Create storage volume.
+	err = s.StoragePoolVolumeCreate()
+	if err != nil {
+		dbStoragePoolVolumeDelete(d.db, req.VolumeName, volumeType, poolID)
+		return InternalError(err)
+	}
+
+	apiEndpoint, err := storagePoolVolumeTypeToApiEndpoint(volumeType)
+	if err != nil {
+		return InternalError(err)
+	}
+	return SyncResponseLocation(true, nil, fmt.Sprintf("/%s/storage-pools/%s/volumes/%s", version.APIVersion, poolName, apiEndpoint))
+}
+
+var storagePoolVolumesTypeCmd = Command{name: "storage-pools/{name}/volumes/{type}", get: storagePoolVolumesTypeGet, post: storagePoolVolumesTypePost}
+
+// /1.0/storage-pools/{pool}/volumes/{type}/{name}
+// Get storage volume of a given volume type on a given storage pool.
+func storagePoolVolumeTypeGet(d *Daemon, r *http.Request) Response {
+	// Get the name of the storage volume.
+	volumeName := mux.Vars(r)["name"]
+
+	// Get the name of the storage pool the volume is supposed to be
+	// attached to.
+	poolName := mux.Vars(r)["pool"]
+
+	// Get the name of the volume type.
+	volumeTypeName := mux.Vars(r)["type"]
+
+	// Convert the volume type name to our internal integer representation.
+	volumeType, err := storagePoolVolumeTypeNameToType(volumeTypeName)
+	if err != nil {
+		return BadRequest(err)
+	}
+	// Check that the storage volume type is valid.
+	if !shared.IntInSlice(volumeType, supportedVolumeTypes) {
+		return BadRequest(fmt.Errorf("Invalid storage volume type %s.", volumeTypeName))
+	}
+
+	// Get the ID of the storage pool the storage volume is supposed to be
+	// attached to.
+	poolID, err := dbStoragePoolGetID(d.db, poolName)
+	if err != nil {
+		return InternalError(err)
+	}
+
+	// Get the storage volume.
+	_, volume, err := dbStoragePoolVolumeGetType(d.db, volumeName, volumeType, poolID)
+	if err != nil {
+		return SmartError(err)
+	}
+
+	volumeUsedBy, err := storagePoolVolumeUsedByGet(d, volume.VolumeName)
+	if err != nil {
+		return InternalError(err)
+	}
+	volume.VolumeUsedBy = volumeUsedBy
+
+	etag := []interface{}{volume.VolumeName, volume.VolumeType, volume.VolumeUsedBy, volume.VolumeConfig}
+
+	return SyncResponseETag(true, volume, etag)
+}
+
+// /1.0/storage-pools/{pool}/volumes/{type}/{name}
+func storagePoolVolumeTypePost(d *Daemon, r *http.Request) Response {
+	return BadRequest(fmt.Errorf("Storage volumes can currently not be renamed."))
+}
```

Comment:
Same as with the pool one. If it's not implemented, then don't implement it. Just remove the function and let the http server do its job.